jsk_arc2017_baxter/euslisp/lib/arc-interface.l

```diff
@@ -279,50 +298,31 @@
                   3000 (send *ri* :get-arm-controller arm) 0) ;; revert baxter
           (send *ri* :wait-interpolation)))
       ;; lift object
-      (ros::ros-info "[:try-to-pick-object-in-bin] arm:~a lift the object" arm)
+      (ros::ros-info "[:try-to-pick-object] arm:~a lift the object" arm)
       (send *ri* :gripper-servo-off arm)
       (send *ri* :angle-vector-raw (send *baxter* arm :move-end-pos #f(0 0 200) :world)
             3000 (send *ri* :get-arm-controller arm) 0)
       (send *ri* :wait-interpolation)
       (unix::sleep 1)  ;; wait for arm to follow
       (setq graspingp (send *ri* :graspingp arm))
-      (ros::ros-info "[:try-to-pick-object-in-bin] arm:~a graspingp: ~a" arm graspingp)
+      (ros::ros-info "[:try-to-pick-object] arm:~a graspingp: ~a" arm graspingp)
     graspingp))
   (:ik->bin-center
-    (arm bin &key (offset #f(0 0 0)) (coords-pitch 0) (rotation-axis t) (use-gripper nil))
-    (let (bin-box bin-coords)
-      (setq bin-box (gethash bin bin-boxes-))
-      (setq bin-coords (send *ri* :tf-pose->coords
-                             (send (send bin-box :header) :frame_id)
-                             (send bin-box :pose)))
-      (send bin-coords :translate
-            (v+ (float-vector
-                  0.0 0.0 (/ (m->mm (send bin-box :dimensions :z)) 2.0))
-                 offset) :world)
-      (send bin-coords :rotate (* pi (/ coords-pitch 180)) :y)
-      (send *baxter* arm :inverse-kinematics bin-coords
-            :rotation-axis rotation-axis
-            :use-gripper use-gripper)))
-
-  (:ik->bin-entrance
     (arm bin &key (offset #f(0 0 0)) (rpy #f(0 0 0)) (rotation-axis t) (use-gripper nil))
-    (let (bin-box bin-coords)
-      (setq bin-box (gethash bin bin-boxes-))
-      (setq bin-coords (send *ri* :tf-pose->coords
-                             (send (send bin-box :header) :frame_id)
-                             (send bin-box :pose)))
-      (send bin-coords :translate
-            (v+ (float-vector
-                  (- (/ (m->mm (send bin-box :dimensions :x)) 2.0))
-                  0.0 (/ (m->mm (send bin-box :dimensions :z)) 2.0))
-                offset) :world)
+    (let (bin-cube bin-coords)
+      (setq bin-cube (gethash bin bin-cubes-))
+      (setq bin-coords (send bin-cube :copy-worldcoords))
       (send bin-coords :rotate (aref rpy 0) :z)
       (send bin-coords :rotate (aref rpy 1) :y)
       (send bin-coords :rotate (aref rpy 2) :x)
+      (send bin-coords :translate
+            (v+ (float-vector
+                  0.0 0.0 (/ (z-of-cube bin-cube) 2.0))
+                 offset) :world)
```

Comment:
"z" of `z-of-cube` is repesented in local coodinates, isn't it?
Then please translate in local.
Offset can be applied in whether local or world.